src/System.Memory/src/System/Buffers/Text/Utf8Formatter.Date.cs

```diff
@@ -0,0 +1,387 @@
+            new byte[] { (byte)'m', (byte)'a', (byte)'r' },
+            new byte[] { (byte)'a', (byte)'p', (byte)'r' },
+            new byte[] { (byte)'m', (byte)'a', (byte)'y' },
+            new byte[] { (byte)'j', (byte)'u', (byte)'n' },
+            new byte[] { (byte)'j', (byte)'u', (byte)'l' },
+            new byte[] { (byte)'a', (byte)'u', (byte)'g' },
+            new byte[] { (byte)'s', (byte)'e', (byte)'p' },
+            new byte[] { (byte)'o', (byte)'c', (byte)'t' },
+            new byte[] { (byte)'n', (byte)'o', (byte)'v' },
+            new byte[] { (byte)'d', (byte)'e', (byte)'c' },
+        };
+
+        private static readonly TimeSpan NullOffset = TimeSpan.MinValue;
+
+        #endregion Constants
+
+        /// <summary>
+        /// Formats a DateTimeOffset as a UTF8 string.
+        /// </summary>
+        /// <param name="value">Value to format</param>
+        /// <param name="buffer">Buffer to receive UTF8 string</param>
+        /// <param name="bytesWritten">Receives the length of the formatted text in bytes</param>
+        /// <param name="format">The standard format to use</param>
+        /// <returns>
+        /// true for success. "bytesWritten" contains the length of the formatted test in bytes.
+        /// false if buffer was too short. Iteratively increase the size of the buffer and retry until it succeeds. 
+        /// </returns>
+        /// <exceptions>
+        /// <cref>System.FormatException</cref> if the format is not valid for this data type.
+        /// </exceptions>
+        public static bool TryFormat(DateTimeOffset value, Span<byte> buffer, out int bytesWritten, StandardFormat format = default)
+        {
+            TimeSpan offset = NullOffset;
+            char symbol = format.Symbol;
+            if (format.IsDefault)
+            {
+                symbol = 'G';
+                offset = value.Offset;
+            }
+
+            switch (symbol)
+            {
+                case 'R':
+                    return TryFormatDateTimeR(value.UtcDateTime, buffer, out bytesWritten);
+
+                case 'l':
+                    return TryFormatDateTimeL(value.UtcDateTime, buffer, out bytesWritten);
+
+                case 'O':
+                    return TryFormatDateTimeO(value.DateTime, value.Offset, buffer, out bytesWritten);
+
+                case 'G':
+                    return TryFormatDateTimeG(value.DateTime, offset, buffer, out bytesWritten);
+
+                default:
+                    throw new FormatException(SR.Argument_BadFormatSpecifier);
+            }
+        }
+
+        /// <summary>
+        /// Formats a DateTime as a UTF8 string.
+        /// </summary>
+        /// <param name="value">Value to format</param>
+        /// <param name="buffer">Buffer to receive UTF8 string</param>
+        /// <param name="bytesWritten">Receives the length of the formatted text in bytes</param>
+        /// <param name="format">The standard format to use</param>
```

Comment:
It would be good to list all supported formats in the docs. 
Also, we need to describe the 'l' format somewhere. It's not on MSDN currently.